src/Snake.DataAccess/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Snake.Game.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace Snake.DataAccess.Repositories
+{
+    public class GenericRepository<TEntity>: IGenericRepository<TEntity> where TEntity : class
+    {
+        internal SnakeDBContext context;
+        internal DbSet<TEntity> dbSet;
+
+        public GenericRepository(SnakeDBContext context)
+        {
+            this.context = context;
+            this.dbSet = context.Set<TEntity>();
+        }
+
+        public virtual IEnumerable<TEntity> Get(
+            Expression<Func<TEntity, bool>> filter = null,
```

Comment:
Default parameters are not used and are exposing implementation details. Are they really needed?